src/Spago/PackageSet.hs

```diff
@@ -219,6 +220,32 @@ localImportPath (Dhall.Import
   })              = Just $ Text.unpack $ pretty localImport
 localImportPath _ = Nothing
 
+-- | In a Monorepo we don't wish to rebuild our shared packages over and over, 
+-- | so we build into an output folder at the highest point
+-- | (where, hopefully, packages.dhall lives)
+findRootOutputPath :: Spago m => System.IO.FilePath -> m (Maybe System.IO.FilePath)
+findRootOutputPath path = do
+  echoDebug "Locating root path of packages.dhall"
+  imports <- liftIO $ Dhall.readImports $ Text.pack path
+  let localImports = catMaybes (map localImportPath imports)
+  pure $ (flip System.FilePath.replaceFileName) "output" <$> (findRootPath localImports)
+
+-- | Given a list of filepaths, find the one with the least folders
+findRootPath :: [System.IO.FilePath] -> Maybe System.IO.FilePath
+findRootPath paths 
+  = foldr comparePaths Nothing paths
+  where
+    isLessThan :: Ord a => a -> Maybe a -> Bool
+    isLessThan a maybeA
+      = isNothing maybeA 
+      || fromMaybe False (fmap (\a' -> a < a') maybeA)
+    
+    comparePaths path current 
+      = if isLessThan 
+          (length (System.FilePath.splitSearchPath path))
+          (length <$> current)
+        then Just path
+        else current
```

Comment:
Ahh right - sounds like we have everything we need to do this nicely. Will give this a smash.

Reply:
Does this implementation work?

Reply:
I would have liked something more like "is this file importing this other one" rather than "is this directory shorter", but let's go with this for now